File: camerax_android/android/src/main/kotlin/dev/zeekr/camerax_android/core/ImageInfoImpl.kt
package dev.zeekr.camerax_android.core

import androidx.camera.core.ImageInfo
import dev.zeekr.camerax_android.CameraXApiPigeonProxyApiRegistrar
import dev.zeekr.camerax_android.FlashStateApi
import dev.zeekr.camerax_android.PigeonApiImageInfoProxyApi

class ImageInfoImpl(registrar: CameraXApiPigeonProxyApiRegistrar) : PigeonApiImageInfoProxyApi(registrar) {
    override fun timestamp(pigeon_instance: ImageInfo): Long {
        return pigeon_instance.timestamp
    }

    override fun flashState(pigeon_instance: ImageInfo): FlashStateApi {
        return pigeon_instance.flashState.flashStateApi
    }

    override fun rotationDegrees(pigeon_instance: ImageInfo): Long {
        return pigeon_instance.rotationDegrees.toLong()
    }
}

File: camerax_android/lib/src/core/image_info_impl.dart
import 'package:camerax_android/src/api.dart';
import 'package:camerax_android/src/core.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class ImageInfoImpl implements ImageInfo {
  final ImageInfoProxyApi api;

  ImageInfoImpl.internal(this.api);

  @override
  FlashState get flashState => api.flashState.impl;
  @override
  int get rotationDegrees => api.rotationDegrees;
  @override
  int get timestamp => api.timestamp;
}

extension ImageInfoProxyApiX on ImageInfoProxyApi {
  ImageInfo get impl => ImageInfoImpl.internal(this);
}
